lib/adyen/rest/authorise_payment.rb

```diff
@@ -185,6 +185,60 @@ def reauthorise_recurring_payment(attributes)
         execute_request(request)
       end
 
+      # The Response class implements some extensions for the list recurring details call.
+      # @see Adyen::REST::Response
+      class ListRecurringDetailsResponse < Adyen::REST::Response
+        # Returns a list of recurring details
+        # @return [Array] A not empty array if there is at least a recurring detail
+        def details
+          list = []
+          index = 0
+
+          while !attributes["recurringDetailsResult.details.{%}.recurringDetailReference".gsub("{%}", index.to_s)].empty? do
+            response = { card: {} }
+            response[:recurring_detail_reference] = attributes["recurringDetailsResult.details.{%}.recurringDetailReference".gsub("{%}", index.to_s)]
+            response[:creation_date]              = attributes["recurringDetailsResult.details.{%}.creationDate".gsub("{%}", index.to_s)]
+            response[:variant]                    = attributes["recurringDetailsResult.details.{%}.variant".gsub("{%}", index.to_s)]
+            response[:card][:holder_name]         = attributes["recurringDetailsResult.details.{%}.card.holderName".gsub("{%}", index.to_s)]
+            response[:card][:expiry_month]        = attributes["recurringDetailsResult.details.{%}.card.expiryMonth".gsub("{%}", index.to_s)]
+            response[:card][:expiry_year]         = attributes["recurringDetailsResult.details.{%}.card.expiryYear".gsub("{%}", index.to_s)]
+            response[:card][:number]              = attributes["recurringDetailsResult.details.{%}.card.number".gsub("{%}", index.to_s)]
```

Comment:
Hi @wvanbergen,

There are two main things I dont like in this implementation. 1) be forced to use a loop in order to get all attributes 2) We dont have a hierarchy in the attributes.  

So, I came up with this new version where I extracted a method to be responsible for map a response list. Then, when we need to parse a new kind of response list we just need to map the attributes of the response we want, and pass to Adyen::REST::Response#map_response_list.

Please, take a look at this gist and tell me what do you think:  https://gist.github.com/rinaldifonseca/438933aeeb1d0ee2ced1

Thanks!